### PR TITLE
Some schema cleanup.

### PR DESCRIPTION
**Pull request by oliverchang**
- Make Purpose section more concise. We don't need a history lesson here anymore :) 
- Remove old changelogs.
- Alphabetically sort databases and ecosystems.
- Add some missing databases (Curl, NVD).

### CODE REVIEW

**Comment by oliverchang** [comment]
CC @andrewpollock

**Comment by andrewpollock** [comment]
@hayleycd may wish to review as well?

**Comment by hayleycd** [comment]
I ran this locally and it looks good to me.

File: docs/schema.md [hunk 1]
```diff
@@ -1528,235 +1542,6 @@ example encoding of a vulnerability entry.
...
-  GitHub (`schema_version`, top-level `database_specific`, `credits`,
-  `severity`, relaxation of version enumeration requirement).
-- 2022-03-24 Released version 1.3.0. Added `last_affected` event type and
-  `database_specific` to `affected[].ranges[]`.
-  Context: https://github.com/ossf/osv-schema/issues/35.
-- 2023-02-21 Released version 1.4.0. Added per package `severity` and
-  credit types.
-- 2023-04-26 Released version 1.5.0. Added new reference types.
-- 2023-08-11 Released version 1.6.0. Several new databases and clarified
-  definitions of `aliases` and `related`.
```
Comment (chrisbloom7) [inline docs/schema.md:1559]: Makes sense to remove this from the schema, but I was surprised to it wasn't replaced with a separate changelog file. Not a blocker, but do you feel it's not worth a log at all?

**Review by chrisbloom7 — APPROVED** [review]
Thanks for adding the sorted tables!

File: docs/schema.md (hunk 1 above)
Comment (oliverchang) [reply docs/schema.md:1559]: Thanks for the callout! I've pulled this out into a separate CHANGELOG.md

**Pull request merged**